tests/components/modbus/test_climate.py

```diff
@@ -101,3 +110,31 @@ async def test_service_climate_update(hass, mock_pymodbus):
         "homeassistant", "update_entity", {"entity_id": entity_id}, blocking=True
     )
     assert hass.states.get(entity_id).state == "auto"
+
+
+async def test_restore_state_climate(hass):
+    """Run test for sensor restore state."""
+
+    climate_name = "test_climate"
+    entity_id = f"{CLIMATE_DOMAIN}.{climate_name}"
+    test_value = State(entity_id, 35)
+    test_value.attributes = {ATTR_TEMPERATURE: 37}
+    config_sensor = {
+        CONF_NAME: climate_name,
+        CONF_TARGET_TEMP: 117,
+        CONF_CURRENT_TEMP: 117,
+    }
+    mock_restore_cache(
+        hass,
+        (test_value,),
+    )
+    await base_config_test(
+        hass,
+        config_sensor,
+        climate_name,
+        CLIMATE_DOMAIN,
+        CONF_CLIMATES,
+        None,
+        method_discovery=True,
+    )
+    assert hass.states.get(entity_id).state == HVAC_MODE_AUTO
```

Comment:
Should we assert the restored temperature?

Reply:
Done